### PR TITLE
Add rust_lint.sh, use it in CI

### PR DESCRIPTION
**Pull request by banool**
### Description
Long requested, much desired, finally here. Running this script will ensure that your PR passes the `rust-lint` step in CI.

### Test Plan
```
scripts/rust_lint.sh
```
```
scripts/rust_lint.sh --check
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4175)

### CODE REVIEW

File: scripts/rust_lint.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# This assumes you have already installed cargo-sort:
+# cargo install cargo-sort
+#
+# The best way to do this however is to run scripts/dev_setup.sh
+
+# Make sure we're in the root of the repo.
+cd "$(dirname "$0")"
```
Comment (rustielin) [inline scripts/rust_lint.sh:9]: nit: usually I'd rather assert that we're at the root, rather than cd to it

**Review by rustielin — APPROVED** [review]
LGTM!!

File: scripts/rust_lint.sh (hunk 1 above)
Comment (banool) [reply scripts/rust_lint.sh:9]: Open to changing this if you feel strongly about it / others agree, idm.

**Review by areshand — APPROVED** [review]

**Review by davidiw — COMMENTED** [review]
forgive my naivete is this duplicating pre-commit or do they do different things?

**Comment by banool** [comment]
We don't do all of this in pre-commit, just fmt. Running xclippy in pre-commit might not be desirable, it's pretty slow. Either way we can consider this in a separate PR.

**Comment by banool** [comment]
- Changed it so it only verifies that you're at the root.
- Added a link to a wrapper that lets you run it from anywhere in the repo.

**Pull request merged**